onnxruntime/contrib_ops/cpu/transformers/sequences.cc

```diff
@@ -0,0 +1,76 @@
+#include "sequences.h"
+
+namespace onnxruntime {
+namespace contrib {
+namespace transformers {
+
+void Sequences::Init(AllocatorPtr allocator, const OrtValue& input_ids, int batch_beam_size, int sequence_length, int max_length) {
+  size_t sequences_size = SafeInt<size_t>(batch_beam_size) * max_length;
+  size_t buffer_size = sequences_size + sequences_size;
+  gsl::span<int64_t> buffer = AllocateBuffer<int64_t>(allocator, sequences_space_buffer_, buffer_size, true, static_cast<int64_t>(0));
+
+  sequences[0] = buffer.subspan(0, sequences_size);
```

Comment:
Are 2 buffers needed? with max_length known or with default max_length - could we append to the same buffer based on last position?

Reply:
For example, currently there are two sequences: "a b",  "c d". When we append next tokens (for example, "e" and "f" are top-2), the best sequences might be "a b e", "a b f" after top-k sorting.

That means we need to keep track of sequences of previous steps ("a b", "c d"), and also current steps ("a b e", "a b f"). One way to do that is to have two buffers, one for previous step, and another for current step.

If we do not use two buffers, we have to record the beam index picked in each step, then reverse trace to find the whole sequences. That will be more complicated.